lib/resolvers/waifutcg.js

```diff
@@ -0,0 +1,157 @@
+'use strict';
+
+import Resolver from '../resolver';
+
+const RARITY_NAMES = ['common', 'uncommon', 'rare', 'super', 'ultra', 'legendary', 'mythical', 'god', 'special', 'promo'];
+
+export default class WaifuTCG extends Resolver {
+	transformURL(url, ctx) {
+		ctx.type = url.pathname.replace('/', '');
+		ctx.tcguser = url.searchParams.get('user');
+		switch (url.pathname) {
+
+			case '/booster':
+				ctx.cache_key = false; // booster Data is highly volatile and should not be cached
+			// eslint-disable-next-line no-fallthrough
+			case '/hand':
+			case '/pullfeed': {
+				ctx.headers = {
+					'Accept': 'application/json'
+				};
+				ctx.parse = 'json';
+			}
+			// eslint-disable-next-line no-fallthrough
+			case '/':
+			case '/profile': // I really thought i had implemented application/json for /profile, but I guess i did not.
+			case '/rules':
+			case '/discord':
+			default: {
+				return url.href;
+			}
+		}
+	}
+
+	processBody(data, mode, ctx) {
+		let title = `${data.user ? data.user : ctx.tcguser}'s Waifu TCG ${ctx.type}`;
+		if (ctx.type === 'pullfeed') {
+			title = 'Waifu TCG Pullfeed';
+		}
+		const returnobject = {
+			'v': 5,
+			'accent': '#8F6AD3D8',
+			'short': {
+				'type': 'header',
+				title,
+				'image': {
+					'type': 'image',
+					'url': 'https://cdn.discordapp.com/icons/536128175399501824/2c63c6179e78e9d4afa4881b3d25f39c.png',
```

Comment:
Is it possible to avoid using Discord-hosted images directly? There have been image hotlinking issues in the past with Discord's CDN.

Reply:
I'll move it to my s3 bucket/cloudfront

Reply:
Resolved with 19082b1